### ISSUE TITLE
27 - FORTIFY - CONNECTED CARS 🚗

### ISSUE THREAD

**Issue by psifrous** [CodeBreakMIT/CodeBreak1.0#29]
## ℹ️ Project information

- **Project Name**: Connected Cars
- **Short Project Description**: _A deployable solution for many major road and traffic problems faced in day to day life._

| Project Modules |  Description |
| --- | --- |
| Congestion Prevention | Computer Vision-based approach to control traffic lights based upon density calculation of vehicles |
| Emergency Vehicle Priority | Notifying vehicles that are driving ahead of the approaching emergency vehicle |
| Drowsiness Detection | Computer Vision-based approach to identify drowsiness of driver |
| Pothole Detection | Using Yolov2 to identify potholes on the road and notify concerned authorities for the fix |

- **Team ID**: 27
- **Team Name**: FORTIFY
- **Team Members**: _Naman Bansal [@psifrous](https://github.com/psifrous), Swapnil Panwala [@panwalas](https://github.com/panwalas), Akshita Mehta [@MehtaAkshita](https://github.com/MehtaAkshita), Arjun Mohnot [@Arjun009](https://github.com/https://github.com/Arjun009)_
- **Demo Link**: _https://www.youtube.com/watch?v=Kq-9fL6Spwc_
- **Presentation Link**: _https://github.com/psifrous/connected-cars/blob/master/Presentation/Connected_Cars.pdf_
- **Repository Link**: _https://github.com/psifrous/connected-cars_
- **Labels**: _ML, Image & Video Processing, Android App, Web Development_

## 🔥 Your Pitch
_Connected Cars is an initiative to improve traffic conditions, shorten the duration of time taken by emergency vehicles to reach the desired destination, avoiding potholes and crashes along with making cities smart in an efficient and effective way by making vehicles communicate i.e. help. Moreover, our concept aims to use the existing infrastructure installed in roads, crossings and cars, thus making it feasible to deploy economically!_

Our project comprises of the following modules:
1) **Congestion Prevention:** Don't you get fed up when you have to wait on traffic signal even when there is no car in the other lane. Can we control traffic lights based on car density in a particular lane? Yes! 
2) **Emergency Vehicle Priority:** Many people die in India because emergency vehicles don't reach the required destination in time. Can we save life by informing people in advance about an incoming emergency vehicle? Definitely yes!
3) **Drowsiness Detection:** Drowsy driving is the dangerous combination of driving and sleepiness or fatigue and it has led to many car accidents. Can we wake up the driver by playing some loud alarm? Super yes!
4) **Pothole Detection**: Potholes can cause severe damages to cars, passengers and to pedestrians too. It can damage your tire's sidewall or belts. Even a minor impact may knock your vehicle out of alignment. A pothole strike can damage your shocks or struts, or harm your suspension. Can we detect them and inform about that to the concerned authority and maintain a database? Sure yes! 

So drive happy and safe 😊  

### Backend
* Yolov2
* OpenCV
* Keras
* TensorFlow
* Flask
### Frontend
* Dash by Plotly
* JavaScript
* HTML
* CSS
* Android Studio
### Database
* Firebase
### API's
* Plate Recognizer API
* Way2Sms API
* Google Maps API

## 🔦 Any other specific thing you want to highlight?
_In our project we have developed 1 Android App, with 4 integrated modules in a web app_

**Current Opportunity** 🚀 :  We can use these modules on existing infrastructure i.e CCTV's installed on public places/traffic signals, and in government vehicles, we don't need any new infrastructure, we will just extract more information from the current infrastructure present.

## ✅ Checklist

**Before you post the issue**:
- [✅] You have followed the issue title format.
- [✅] You have mentioned the correct labels.
- [✅] You have provided all the information correctly.
